build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,210 @@
+			}
+
+			const configOptions = ExtractorConfig.tryLoadForFolder({
+				startingFolder: pkg.directory,
+			});
+			if (configOptions === undefined) {
+				this.verbose(`No api-extractor config found for ${pkg.name}. Skipping.`);
+				return;
+			}
+			const extractorConfig = ExtractorConfig.prepare(configOptions);
+
+			const config: ReleaseTagConfig = {
+				pkg,
+				extractorConfig,
+				releaseType: this.flags.types as ReleaseTag,
+				json,
+			};
+
+			packageUpdate = updatePackageJsonTypes(config, this.logger);
+
+			updatePackageLists(this.packageList, packageUpdate, pkg);
+		});
+	}
+
+	public async run(): Promise<PackageTypesList> {
+		// Calls processPackage on all packages.
+		await super.run();
+
+		const flags = this.flags;
+
+		if (this.packageList.packagesUpdated.length === 0) {
+			this.log(`No updates in package.json for ${flags.types} release tag`);
+		}
+
+		return this.packageList;
+	}
+}
+
+/**
+ * Updates the types/typing field in package.json.
+ * @param config - An interface specifying the package, extractor config, release type, JSON data
+ * @param log - The logger used for logging verbose information.
+ * @returns true if the update was successful, false otherwise.
+ */
+function updatePackageJsonTypes(config: ReleaseTagConfig, log: CommandLogger): boolean {
+	let packageUpdate = false;
+	try {
+		if (config.extractorConfig.rollupEnabled === true) {
+			switch (config.releaseType) {
+				case "alpha": {
+					config.json.types = path.relative(
+						config.pkg.directory,
+						config.extractorConfig.alphaTrimmedFilePath,
+					);
+					packageUpdate = true;
+					break;
+				}
+
+				case "beta": {
+					config.json.types = path.relative(
+						config.pkg.directory,
+						config.extractorConfig.betaTrimmedFilePath,
+					);
+					packageUpdate = true;
+					break;
+				}
+
+				case "public": {
+					config.json.types = path.relative(
+						config.pkg.directory,
+						config.extractorConfig.publicTrimmedFilePath,
+					);
+					packageUpdate = true;
+					break;
+				}
+
+				default: {
+					config.json.types = path.relative(
+						config.pkg.directory,
+						config.extractorConfig.untrimmedFilePath,
+					);
+					packageUpdate = true;
+					break;
+				}
+			}
+		}
+	} catch {
+		log.verbose(`Unable to update types: ${JSON.stringify(config.extractorConfig)}`);
+	}
+	return packageUpdate;
+}
+
+/**
+ * Updates a package types list based on whether a package was updated or not.
+ *
+ * @param packageList - The package types list to be updated.
+ * @param packageUpdate - A boolean indicating whether the package was updated.
+ * @param pkg - The package being processed.
+ * @returns The updated package types list.
+ */
+function updatePackageLists(
+	packageList: PackageTypesList,
+	packageUpdate: boolean,
+	pkg: Package,
+): PackageTypesList {
+	if (packageUpdate) {
+		packageList.packagesUpdated.push(pkg.name);
+	} else {
+		packageList.packagesNotUpdated.push(pkg.name);
+	}
+	return packageList;
+}
```

Comment:
This is simple enough it can just be inlined. It doesn't affect the testability of any of the other code.